src/OpenLoco/CMakeLists.txt

```diff
@@ -0,0 +1,651 @@
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Localisation/FormatArguments.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Location.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Map/Map.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Map/TileLoop.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Math/Bound.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Math/Trigonometry.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Math/Vector.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Speed.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Types.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Ui/Types.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Utility/Collection.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Utility/Exception.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Utility/Numeric.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Utility/Prng.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Utility/Stream.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Utility/String.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Viewport.hpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/ZoomLevel.hpp")
+
+if (APPLE)
+    set(OLOCO_MM_SOURCES 
+        "${CMAKE_CURRENT_SOURCE_DIR}src/Platform/Platform.Macos.mm")
+    set_source_files_properties(${OLOCO_MM_SOURCES} PROPERTIES COMPILE_FLAGS "-x objective-c++")
+endif()
+
+# Disable optimizations for interop.cpp for all compilers, to allow optimized
+# builds without need for -fno-omit-frame-pointer
+if (MSVC)
+    set_source_files_properties("${CMAKE_CURRENT_SOURCE_DIR}/src/Interop/Interop.cpp" PROPERTIES COMPILE_FLAGS "/Oy- /Od")
+else ()
+    set_source_files_properties("${CMAKE_CURRENT_SOURCE_DIR}/src/Interop/Interop.cpp" PROPERTIES COMPILE_FLAGS "-fno-omit-frame-pointer -O0")
+endif ()
+
+if (MINGW)
+    loco_add_library(OpenLoco SHARED
+        PRIVATE_FILES
+            ${OLOCO_SOURCES})
+    target_link_libraries(OpenLoco winmm)
+else ()
+    if (MSVC)
+        # Headers added as well so that shown correctly in IDEs
+        loco_add_library(OpenLoco SHARED 
+            PRIVATE_FILES
+                ${OLOCO_SOURCES}
+                ${OLOCO_HEADERS})
+    else ()
+        loco_add_executable(OpenLoco 
+            PRIVATE_FILES
+                ${OLOCO_SOURCES}
+                ${OLOCO_MM_SOURCES})
+        add_dependencies(OpenLoco linkable_sections)
+        # set_source_files_properties has to be set in the CMakeLists that uses them
+        set_source_files_properties(
+            ${LOCO_SECTIONS}
+            PROPERTIES
+            EXTERNAL_OBJECT true
+            GENERATED true
+        )
+        target_sources(OpenLoco
+            PRIVATE
+                ${LOCO_SECTIONS})
+        set_target_properties(OpenLoco PROPERTIES LINK_FLAGS ${LOCO_SEGMENT_LINKER_FLAGS})
+
+        if (TARGET SDL2::SDL2main)
+            target_link_libraries(OpenLoco SDL2::SDL2main)
+        endif ()
+    endif ()
+endif ()
+
+target_link_libraries(OpenLoco
+    ${SDL2_LIB}
+    Threads::Threads
+    yaml-cpp::yaml-cpp
+    nonstd::span-lite
+    ghcFilesystem::ghc_filesystem
```

Comment:
we use compiler's filesystem whenever possible, don't we? Shouldn't this depend on whether we need it? There's even already code that conditionally links `stdc++fs` down below

Reply:
Thats a good point i thought we needed it for some linux builds but we don't actually need it at all. So, time to delete.